com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -22,151 +24,105 @@ public SwizzleNode()
-        }
-
-        static Dictionary<TextureChannel, string> s_ComponentList = new Dictionary<TextureChannel, string>
-        {
-            {TextureChannel.Red, "r" },
-            {TextureChannel.Green, "g" },
-            {TextureChannel.Blue, "b" },
-            {TextureChannel.Alpha, "a" },
-        };
-
         [SerializeField]
-        TextureChannel m_RedChannel;
+        string _maskInput = "xyzw";
 
-        [ChannelEnumControl("Red Out")]
-        public TextureChannel redChannel
+        [TextControl("Mask:")]
+        public string maskInput
         {
-            get { return m_RedChannel; }
+            get { return _maskInput; }
             set
             {
-                if (m_RedChannel == value)
+                if (_maskInput.Equals(value))
                     return;
-
-                m_RedChannel = value;
-                Dirty(ModificationScope.Node);
+                _maskInput = value;
+                UpdateNodeAfterDeserialization();
+                owner.ValidateGraph();
+                Dirty(ModificationScope.Topological);
             }
         }
 
-        [SerializeField]
-        TextureChannel m_GreenChannel;
-
-        [ChannelEnumControl("Green Out")]
-        public TextureChannel greenChannel
+        //1.mask(xyzw) 0< length <=4
+        //2.mask cant be character other than "xyzw"
+        //3.If the input is not valid, wont genetate shader code and give errors
+        public bool ValidateMaskInput(int InputValueSize)
         {
-            get { return m_GreenChannel; }
-            set
+            bool MaskInputIsValid = true;
+            char[] MaskChars = _maskInput.ToCharArray();
+            char[] AllChars  = {'x', 'y' , 'z', 'w'};
+            List<char> CurrentChars = new List<char>();
+            for (int i = 0; i < InputValueSize; i++)
             {
-                if (m_GreenChannel == value)
-                    return;
-
-                m_GreenChannel = value;
-                Dirty(ModificationScope.Node);
+                CurrentChars.Add(AllChars[i]);
             }
-        }
 
-        [SerializeField]
-        TextureChannel m_BlueChannel;
-
-        [ChannelEnumControl("Blue Out")]
-        public TextureChannel blueChannel
-        {
-            get { return m_BlueChannel; }
-            set
+            foreach (char c in MaskChars)
             {
-                if (m_BlueChannel == value)
-                    return;
-
-                m_BlueChannel = value;
-                Dirty(ModificationScope.Node);
+                if (!CurrentChars.Contains(c))
+                {
+                    MaskInputIsValid = false;
+                }
             }
-        }
-
-        [SerializeField]
-        TextureChannel m_AlphaChannel;
-
-        [ChannelEnumControl("Alpha Out")]
-        public TextureChannel alphaChannel
-        {
-            get { return m_AlphaChannel; }
-            set
+            if (MaskChars.Length == 0 || MaskChars.Length > 4)
```

Comment:
Even though it's not strictly necessary for Length, I generally like to do tests that will catch errors for any possible value.  So in this case:
if (MaskChars.Length <= 0 || MaskChars.Length > 4)